test/python/test_qobj.py

```diff
@@ -28,6 +32,24 @@ def test_create_qobj(self):
             ]
         )
 
+        self.assertTrue(
+            all(getattr(qobj, required_arg) is not None for required_arg in Qobj.REQUIRED_ARGS))
+
+    def test_as_dict(self):
+        """Test conversion to dict of a Qobj based on the individual elements."""
+        qobj = Qobj(
+            id='12345',
+            header={},
+            config=QobjConfig(shots=1024, memory_slots=2, max_credits=10),
+            experiments=[
+                QobjExperiment(instructions=[
+                    QobjInstruction(name='u1', qubits=[1], params=[0.4]),
+                    QobjInstruction(name='u2', qubits=[1], params=[0.4, 0.2])
+                ])
+            ]
+        )
+        qobj._version = '67890'  # private member variables shouldn't appear in the dict
```

Comment:
I've kept a single test of this in ```test_as_dict``` because it can't be independently tested due to Qobj requiring certain args upon initialization; thus, as_dict will always contain more information than just the ```_version```.